wps/core/repository.py

```diff
@@ -133,16 +148,24 @@ async def get(
         try:
             work_package = await self._dao.get_by_id(work_package_id)
         except ResourceNotFoundError as error:
-            raise self.WorkPackageAccessError from error
+            raise self.WorkPackageAccessError("Work package not found") from error
         if (
             check_valid
             and not work_package.created <= now_as_utc() <= work_package.expires
         ):
-            raise self.WorkPackageAccessError
+            raise self.WorkPackageAccessError("Workpackage has expired")
         if work_package_access_token and work_package.token_hash != hash_token(
             work_package_access_token
         ):
-            raise self.WorkPackageAccessError
+            raise self.WorkPackageAccessError("Invalid work package access token")
+        if (
+            check_valid
+            and work_package.type == WorkType.DOWNLOAD
+            and not await self._access.check_download_access(
+                work_package.user_id, work_package.dataset_id
+            )
+        ):
+            raise self.WorkPackageAccessError("Access has been revoked")
```

Comment:
Could have not existed in the first place.

```suggestion
            raise self.WorkPackageAccessError("No access to dataset.")
```

Reply:
Actually no, the work package can only be created when access exists.